projects/ngneat/helipopper/src/lib/tippy.directive.ts

```diff
@@ -189,6 +215,35 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
     this.viewRef = null;
   }
 
+  /**
+   * This method is useful when you append to an element that you might remove from the DOM.
+   * In such cases we want to hide the tooltip and let it go through the destroy lifecycle.
+   */
+  observeHostVisibility() {
+    if (this.props.appendTo) {
```

Comment:
We'll probably never remember why we need this if statement

Reply:
`// We don't want to observe the host visibility if we are appending to the body.`
Added.
is it clear enough?